Velaptor/UI/ControlBase.cs

```diff
@@ -0,0 +1,209 @@
+        {
+            get => Position.X + Width;
+            set => Position = new Point(value - Width, Position.Y);
+        }
+
+        /// <summary>
+        /// <inheritdoc cref="IControl"/>
+        /// </summary>
+        public virtual int Top
+        {
+            get => Position.Y;
+            set => Position = new Point(Position.X, value);
+        }
+
+        /// <summary>
+        /// <inheritdoc cref="IControl"/>
+        /// </summary>
+        public virtual int Bottom
+        {
+            get => Position.Y + Height;
+            set => Position = new Point(Position.X, value - Height);
+        }
+
+        /// <summary>
+        /// <inheritdoc cref="IControl"/>
+        /// </summary>
+        public virtual int Width { get; set; }
+
+        /// <summary>
+        /// <inheritdoc cref="IControl"/>
+        /// </summary>
+        public virtual int Height { get; set; }
+
+        /// <summary>
+        /// <inheritdoc cref="IControl"/>
+        /// </summary>
+        public virtual bool Visible { get; set; } = true;
+
+        /// <summary>
+        /// <inheritdoc cref="IControl"/>
+        /// </summary>
+        public virtual bool Enabled { get; set; } = true;
+
+        /// <summary>
+        /// Gets a value indicating whether gets a value indicating if the mouse is hovering over the button.
+        /// </summary>
+        public bool IsMouseOver { get; private set; }
+
+        /// <summary>
+        /// <inheritdoc cref="IControl"/>
+        /// </summary>
+        public bool IsLoaded { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the color to tint the control surface.
+        /// </summary>
+        protected Color TintColor { get; set; } = Color.White;
+
+        /// <summary>
+        /// <inheritdoc cref="IControl"/>
+        /// </summary>
+        [ExcludeFromCodeCoverage]
+        public virtual void LoadContent()
+        {
+            if (IsLoaded)
+            {
+                return;
+            }
+
+            IsLoaded = true;
+        }
+
+        /// <inheritdoc cref="IControl"/>
+        public virtual void Update(FrameTime frameTime)
+        {
+            if (IsLoaded is false || Enabled is false)
+            {
+                return;
+            }
+
+            this.currentMouseState = this.mouse.GetState();
+            this.currentMousePos = this.currentMouseState.GetPosition().ToPoint();
+            var controlRect = new Rectangle(Position.X, Position.Y, Width, Height);
+
+            IsMouseOver = controlRect.Contains(this.currentMouseState.GetX(), this.currentMouseState.GetY());
+
+            // If the mouse button is in the down position
+            if (this.currentMouseState.IsLeftButtonDown() && IsMouseOver)
+            {
+                TintColor = Color.FromArgb(255, 190, 190, 190);
+            }
+            else
+            {
+                TintColor = IsMouseOver ? Color.FromArgb(255, 230, 230, 230) : Color.White;
```

Comment:
Take the color here at this line of code and turn it into a property named `MouseHoverColor` with a default value of this color.

This will give the user the ability to customize the color that is shown when the mouse is over the control.